driver-scala/src/test/scala/org/mongodb/scala/internal/UnitObservableSpec.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mongodb.scala.internal
+
+import org.mongodb.scala.{ BaseSpec, Observer, Subscription }
+
+class UnitObservableSpec extends BaseSpec {
```

Comment:
Lets add a chainability case to ensure this actually fixes the reported issue.

```
  it should "work with for comprehensions" in {
    val h = for {
      _ <- UnitObservable(TestObservable(1 to 2))
      _ <- UnitObservable(TestObservable(20 to 30))
    } yield List(1, 2, 3)

    val results = ArrayBuffer[Int]()
    var completed = false
    h.subscribe((s: List[Int]) => results.addAll(s), (t: Throwable) => t, () => completed = true)
    results should equal(List(1, 2, 3))
    completed should equal(true)
  }
```

Reply:
Thank you, done.